coalib/core/Core.py

```diff
@@ -212,8 +212,8 @@ def _schedule_bears(self, bears):
         """
         for bear in bears:
             if self.dependency_tracker.get_dependencies(
-                    bear):  # pragma: no cover
-                logging.warning(
+                    bear):
+                raise RuntimeError(  # Unreachable code
```

Comment:
A lot of that already exists.  We certainly should not rely on no exceptions occurring inside any function; never know when/where one will occur (i.e. `KeyboardInterrupt`).
How about this commit on top
https://github.com/jayvdb/coala/commit/76afe4855dd0e4ccc8aa3f5f5e9ca4a7f47e1299